### ISSUE TITLE
Training loss is not being logged for pytorch-lightning version greater 0.8.5

### ISSUE THREAD

**Issue by sneakyPad** [wandb/client#1320]
Hi,
when using wandb with pytorch-lightning I experienced that the training loss is not automatically logged anymore. I tried the versions > 0.8.5. The test loss is still being fetched. Since this problem also affects other logging integrations, e.g. Neptune.ai (verified), I am not sure on which side the logging is failing. 

Tried code:
```python
from pytorch_lightning.loggers import WandbLogger
wandb.init(project="project")
trainer = pl.Trainer(max_epochs=5, logger=wandb_logger, gpus=0, weights_summary='full')
```
Happened to anyone else?

**Comment by tyomhak** [comment]
Hi @sneakyPad , thanks for flagging this issue. We're looking into this.

**Comment by borisdayma** [comment]
@sneakyPad Do you have a W&B run you can share?

Recently the logging in pytorch lightning has changed where they don't use a dict anymore but directly call `self.log`.
See [the documentation](https://pytorch-lightning.readthedocs.io/en/stable/logging.html#logging-from-a-lightningmodule).

It should have printed a warning in your console mentioning this deprecation, unless your problem is related to something else.

**Comment by ariG23498** [comment]
Hey @sneakyPad 
We are closing this issue due to inactivity. As @borisdayma has said, logging in pytorch lightning has changed, it would be great if you use it and report back to us.
Please feel free to reopen this issue if the ticket still persists.

**Comment by FraPochetti** [comment]
Hi @sneakyPad have you been able to figure this out?
I am stumbling upon the same issue. The train loss is not logged. Everything else is, on the contrary.

@borisdayma the link to the docs you provide is broken.
What were you referring to when you mentioned the following?
> Recently the logging in pytorch lightning has changed where they don't use a dict anymore but directly call self.log.

Thanks eveybody for the help!

**Comment by sneakyPad** [comment]
Hi @FraPochetti!
Sorry @borisdayma for not getting back at this. I remember that I tried `self.log` but at that time I was logging a lot, and if I recall correctly, I was only able to log a single variable with it? (Not quite sure) 

@FraPochetti I ended up with the following:

```Python
tensorboard_logs = {'train_loss': batch_loss} 

return {'loss': batch_loss, 'log': tensorboard_logs}
```

I think the `loss` in the return statement never got logged and I was just a bit careless. I think the trick was using `tensorboard_logs` for `log`. Let me know if this works.

You can also have a look at the full code here: [github.com/sneakyPad/decoding-latent-space-rs/blob/master/models/movies_vae.py](https://github.com/sneakyPad/decoding-latent-space-rs/blob/master/models/movies_vae.py)

**Comment by FraPochetti** [comment]
Thanks a lot!

**Comment by borisdayma** [comment]
I'm not sure which version you use but the pattern for logging is now with `self.log` (see [the docs](https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html#automatic-logging))

**Comment by gillan-krishna** [comment]
Has this issue been resolved? My neptune logger does not log training_step metrics, but does log validation_step metrics, when I make a LightningDataModule to pass the data

I'm using this in my LightningModule
`    def training_step(self, batch, batch_idx):
        x, y = batch
        x_hat = self.model(x)
        loss = self.loss_fn(x_hat, y)
        fs = fscore(x_hat, y)
        self.log("train/loss", loss, prog_bar=True)
        self.log("train/fscore", fs, prog_bar=True)
        return loss
    
    def validation_step(self, batch, batch_idx):
        x, y = batch
        x_hat = self.model(x)
        loss = self.loss_fn(x_hat, y)
        fs = fscore(x_hat, y)
        self.log("val/loss", loss, prog_bar=True)
        self.log("val/fscore", fs, prog_bar=True)
        return loss`

and   
`  neptune_logger = NeptuneLogger(
    project="username/project",
    api_key=<api_key>,
    log_model_checkpoints=False
    )
    trainer = pl.Trainer(logger=neptune_logger)
    trainer.fit(model=model, datamodule=dm)
`

Curiously, it works if I defined train and val loaders separately and call as 
`trainer.fit(model=model, train_dataloaders=train_loader, val_dataloaders=val_loader)`

**Comment by paraschopra** [comment]
I have the same issue as @gillan-krishna
